packages/commons/src/config/featureFlagsConfig.ts

```diff
@@ -1,20 +1,87 @@
+import { notFound } from "pagopa-interop-models";
 import { z } from "zod";
 
-export const FeatureFlagsConfig = z
+export const FeatureFlagSignalhubWhitelistConfig = z
   .object({
     FEATURE_FLAG_SIGNALHUB_WHITELIST: z
       .enum(["true", "false"])
       .default("false")
-      .transform((value) => value === "true"),
+      .transform((value) => value === "true")
+      .optional(),
     SIGNALHUB_WHITELIST_PRODUCER: z
       .string()
       .transform((value) => value.split(","))
       .pipe(z.array(z.string().uuid()))
       .optional(),
   })
   .transform((c) => ({
-    featureFlagSignalhubWhitelist: c.FEATURE_FLAG_SIGNALHUB_WHITELIST,
+    featureFlagSignalhubWhitelist: c.FEATURE_FLAG_SIGNALHUB_WHITELIST ?? false,
     signalhubWhitelistProducer: c.SIGNALHUB_WHITELIST_PRODUCER,
   }));
 
-export type FeatureFlagsConfig = z.infer<typeof FeatureFlagsConfig>;
+export type FeatureFlagSignalhubWhitelistConfig = z.infer<
+  typeof FeatureFlagSignalhubWhitelistConfig
+>;
+
+export const FeatureFlagAgreementApprovalPolicyUpdateConfig = z
+  .object({
+    FEATURE_FLAG_AGREEMENT_APPROVAL_POLICY_UPDATE: z
+      .enum(["true", "false"])
+      .default("false")
+      .transform((value) => value === "true")
+      .optional(),
+  })
+  .transform((c) => ({
+    featureFlagAgreementApprovalPolicyUpdate:
+      c.FEATURE_FLAG_AGREEMENT_APPROVAL_POLICY_UPDATE ?? false,
+  }));
+
+export type FeatureFlagAgreementApprovalPolicyUpdateConfig = z.infer<
+  typeof FeatureFlagAgreementApprovalPolicyUpdateConfig
+>;
+
+type FeatureFlags = FeatureFlagSignalhubWhitelistConfig &
+  FeatureFlagAgreementApprovalPolicyUpdateConfig;
+
+export type FeatureFlagKeys = keyof Pick<
+  FeatureFlags,
+  {
+    [K in keyof (FeatureFlagSignalhubWhitelistConfig &
+      FeatureFlagAgreementApprovalPolicyUpdateConfig)]: K extends `featureFlag${string}`
+      ? K
+      : never;
+  }[keyof (FeatureFlagSignalhubWhitelistConfig &
+    FeatureFlagAgreementApprovalPolicyUpdateConfig)]
+>;
```

Comment:
I have a doubt here in terms of readability 🧐, 
the number of ad-hoc config type definitions could grow in future, so @sandrotaje what do you think if we create a new type to collect all configs unions?
It increase readability and avoid definitions duplicated in line 53 and 49.

For example we can using `FeatureFlagAllConfig` ?

```suggestion
type FeatureFlagAllConfig = FeatureFlagSignalhubWhitelistConfig & FeatureFlagAgreementApprovalPolicyUpdateConfig;

export type FeatureFlagKeys = keyof Pick<
  FeatureFlags,
  {
    [K in keyof FeatureFlagAllConfig]: K extends `featureFlag${string}`
      ? K
      : never;
  }[keyof FeatureFlagAllConfig]
>;
```